.github/workflows/test-plugins-against-cockpit.yml

```diff
@@ -157,32 +182,107 @@ jobs:
           env: C8Y_CTRL_MODE=mocking,C8Y_CTRL_PROVIDER_VERSION=${{ env.VERSION }}
 
       - name: Upload cypress screenshots
-        if: failure() && env.skip_job != 'true'
+        if: failure()
         uses: actions/upload-artifact@v4
         with:
           retention-days: 5
           name: cypress-screenshots
           path: cypress/screenshots
 
       - name: Upload cypress videos
-        if: always() && env.skip_job != 'true'
+        if: always()
         uses: actions/upload-artifact@v4
         with:
           retention-days: 5
           name: cypress-videos
           path: cypress/videos
 
       - name: Create or update JSON file
-        if: ${{ env.skip_job != 'true' && github.event_name == 'schedule' }}
+        if: ${{ github.event_name == 'schedule' }}
         run: |
+          mkdir -p ./cache-files
           file="./cache-files/${{ env.TAG }}-${{ env.PLUGINS_VERSION }}.json"
           now=$(date --utc +"%Y-%m-%dT%H:%M:%SZ")
           jq -n --arg shellVersion "${{ env.VERSION }}" --arg pluginsVersion "${{ env.PLUGINS_VERSION }}" --arg lastSuccess "$now" '{shellVersion: $shellVersion, pluginsVersion: $pluginsVersion, lastSuccess: $lastSuccess}' > $file
           cat $file
 
-      - name: Upload updated JSON file to cache
-        if: ${{ env.skip_job != 'true' && github.event_name == 'schedule' }}
-        uses: actions/cache@v4
+      - name: Upload updated JSON file
+        if: ${{ github.event_name == 'schedule' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: updated-json-${{ matrix.version_data.tag }}-${{ matrix.version_data.version }}-${{ env.PLUGINS_VERSION }}
+          path: ./cache-files/*.json
+
+  cache-upload:
+    permissions:
+      actions: write
+    if: ${{ github.event_name == 'schedule' && needs.collect-shell-versions.outputs.shell_versions != '[]' }}
+    needs: [collect-shell-versions, run-tests-against-shell]
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
```

Comment:
Seems like there is no need to checkout the repo in this job?

Reply:
it is- without it, line `gh actions-cache delete test-plugins-against-shell-e2e-results --confirm` throws an error 
https://github.com/SoftwareAG/cumulocity-community-plugins/actions/runs/10054589119/job/27789622927